src/System.Runtime.InteropServices.RuntimeInformation/src/RuntimeInformation.Unix.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+
+namespace System.Runtime.InteropServices
+{
+    public static partial class RuntimeInformation
+    {
+        private static string s_osDescription = null;
+        private static Architecture? s_osArch = null;
+
+        public static string OSDescription
+        {
+            get
+            {
+                if (null == s_osDescription)
+                {
+                    s_osDescription = Interop.Sys.GetUnixVersion();
+                }
+
+                return s_osDescription;
+            }
+        }
+
+        public static Architecture OSArchitecture
+        {
+            get
+            {
+                if (null == s_osArch)
```

Comment:
This is not thread safe - assignments of Nullable<T> are not thread safe. You either need to use locks, or you need to use a special placeholder value, like (Architecture)-1 to signal non-initialized state.